.github/workflows/build-test-e2e.yml

```diff
@@ -0,0 +1,52 @@
+name: 'Build/Run End-to-End Tests'
+
+on: push
+jobs:
+  build-and-run-e2e:
+    if: ${{ github.ref == 'develop' || github.ref == 'release-**' || github.actor == github.repository_owner }}
+    runs-on: macos-latest
+    env:
+      CONTAINER_NAME: 'swc-tomcat'
+      CI: true
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup docker (missing on MacOS)
+        if: runner.os == 'macos'
+        run: |
+          # workaround for https://github.com/actions/runner-images/issues/8104
+          export HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK=1
+          brew update
+          brew reinstall qemu
+
+          brew install docker
+          colima start
```

Comment:
Docker is installed by default on Linux runners, but [not on macOS runners](https://stackoverflow.com/a/76490257), so install it here.